### PR TITLE
Add cost range

### PR DESCRIPTION
**Pull request by katherinelc321**
Adds the option to get cost from a date range.[ OSD-7822](https://issues.redhat.com/browse/OSD-7822)

### CODE REVIEW

**Review by drpaneas — APPROVED** [review]
/lgtm

File: cmd/cost/get.go [hunk 1]
```diff
@@ -23,48 +24,46 @@ func newCmdGet(streams genericclioptions.IOStreams) *cobra.Command {
...
+	getCmd.Flags().StringVar(&ops.start, "start", "", "set start date range")
+	getCmd.Flags().StringVar(&ops.end, "end", "", "set end date range")
 	getCmd.Flags().BoolVar(&ops.csv, "csv", false, "output result as csv")
 
-	if err := getCmd.MarkFlagRequired("ou"); err != nil {
-		log.Fatalln("OU flag:", err)
+	return getCmd
+}
+
+func (o *getOptions) complete(cmd *cobra.Command, _ []string) error {
```
Comment (fahlmant) [inline cmd/cost/get.go:41]: This function name is not very clear to me. Is there a reason its named complete and not something like "checkArgs"? Could you add some comments to this as well to explain what this function does and failure case?

File: cmd/cost/get.go [hunk 2]
```diff
@@ -179,10 +205,20 @@ func getOUsRecursive(OU *organizations.OrganizationalUnit, awsClient awsprovider
 }
 
 //Get cost of given account
-func getAccountCost(accountID *string, unit *string, awsClient awsprovider.Client, timePtr *string, cost *float64) error {
+func (o *getOptions) getAccountCost(accountID *string, unit *string, awsClient awsprovider.Client, cost *float64) error {
+
+	var start, end, granularity string
+	if o.time != "" {
```
Comment (fahlmant) [inline cmd/cost/get.go:211]: Is time mutually exclusive with start and end? If all three are set,  as far as I can read this, the start, end and granularity set here (lines 212-213) will be overwritten (lines 2127-219).

File: cmd/cost/list.go [hunk 3]
```diff
@@ -45,11 +48,33 @@ func newCmdList(streams genericclioptions.IOStreams) *cobra.Command {
 	return listCmd
 }
 
+func (o *listOptions) complete(cmd *cobra.Command, _ []string) error {
```
Comment (fahlmant) [inline cmd/cost/list.go:51]: Same questions here as the other `complete`

File: cmd/cost/list.go [hunk 4]
```diff
@@ -61,7 +86,7 @@ func newListOptions(streams genericclioptions.IOStreams) *listOptions {
 }
 
 //List the cost of each OU under given OU
-func listCostsUnderOU(OU *organizations.OrganizationalUnit, awsClient awsprovider.Client, ops *listOptions) error {
+func (o *getOptions) listCostsUnderOU(OU *organizations.OrganizationalUnit, awsClient awsprovider.Client, ops *listOptions) error {
```
Comment (fahlmant) [inline cmd/cost/list.go:89]: Did you mean to use ListOptions here?

File: cmd/cost/get.go (hunk 1 above)
Comment (katherinelc321) [reply cmd/cost/get.go:41]: Refactoring this to make it more clear

File: cmd/cost/get.go (hunk 2 above)
Comment (katherinelc321) [reply cmd/cost/get.go:211]: The function from before is a check that errors out if the 3 flags are provided. So yes time and start/end are mutually exclusive

File: cmd/cost/list.go (hunk 4 above)
Comment (katherinelc321) [reply cmd/cost/list.go:89]: the func listCostsUnderOU calls a func from get.go. I was wanting it to inherit the struct from getOptions. Refactoring this to make it more readable

**Review by dkeohane — APPROVED** [review]
Changes look good - do you need to add documentation to the README?

**Pull request merged**